File: 1337code/208.py
class Trie:
    class Node:
        def __init__(self, value, children):
            self.value = value
            self.children = children
    
    def __init__(self):
        self.root = self.Node('', {})

    def insert(self, word: str) -> None:
        curr = self.root

        for c in word:
            if c in curr.children:
                curr = curr.children[c]
            else:
                next = self.Node(c, {})
                curr.children[c] = next
                curr = next

        if '.' not in curr.children:
            curr.children['.'] = self.Node('.', {})            

    def search(self, word: str) -> bool:
        curr = self.root

        for c in word:
            if c in curr.children:
                curr = curr.children[c]
            else:
                return False
    
        return '.' in curr.children

    def startsWith(self, prefix: str) -> bool:
        curr = self.root

        for c in prefix:
            if c in curr.children:
                curr = curr.children[c]
            else:
                return False

        return True


# Your Trie object will be instantiated and called as such:
word = "apple"
obj = Trie()
obj.insert(word)
param_2 = obj.search(word)
param_3 = obj.startsWith(prefix)

File: 1337code/208.cpp
#include<string.h>
#include<unordered_map>

class Trie {
public:

    Trie() {
        
    }
    
    void insert(string word) {
        
    }
    
    bool search(string word) {
        
    }
    
    bool startsWith(string prefix) {
        
    }

    class Node {
    public:
        int val;
        unordered_map<char, Node> children;

        Node(int val, unordered_map<char) {
            
        }
    }
};

/**
 * Your Trie object will be instantiated and called as such:
 * Trie* obj = new Trie();
 * obj->insert(word);
 * bool param_2 = obj->search(word);
 * bool param_3 = obj->startsWith(prefix);
 */